src/solver/pad_constant/pad_constant_fwd_contiguous.cpp

```diff
@@ -0,0 +1,133 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "miopen/conv_solution.hpp"
+#include "miopen/datatype.hpp"
+#include "miopen/execution_context.hpp"
+#include "miopen/kernel_build_params.hpp"
+#include "miopen/miopen.h"
+#include "miopen/mlo_internal.hpp"
+#include "miopen/pad_constant/solvers.hpp"
+#include "miopen/pad_constant/invoke_params.hpp"
+#include "miopen/tensor_view_5d.hpp"
+
+namespace miopen {
+namespace solver {
+namespace pad_constant_fwd_contiguous {
+bool PadConstantFwdContiguous::IsApplicable(
+    [[maybe_unused]] const ExecutionContext& context,
+    const miopen::pad_constant_fwd_contiguous::ProblemDescription& problem) const
+{
+    if(!problem.IsSameType())
+    {
+        return false;
+    }
+
+    return true;
+}
+
+ConvSolution PadConstantFwdContiguous::GetSolution(
+    [[maybe_unused]] const ExecutionContext& context,
+    const miopen::pad_constant_fwd_contiguous::ProblemDescription& problem) const
+{
+    auto result = ConvSolution{miopenStatusSuccess};
+    auto ydims  = problem.GetYDesc().GetLengths();
+
+    auto input_dtype  = miopen::GetDataType(problem.GetXDesc().GetType());
+    auto output_dtype = miopen::GetDataType(problem.GetYDesc().GetType());
+
+    // for xgridsize: 5d -> 1d
+    size_t output_size = problem.GetYDesc().GetElementSize();
+
+    size_t xlocalsize = 1024;
+    // AlignUp blows up, because output_size can be > int_max. Lovely.
+    // size_t xgridsize  = AlignUp(output_size, xlocalsize);
+    size_t xgridsize =
+        (((static_cast<size_t>(output_size) + xlocalsize - 1) / xlocalsize) * xlocalsize);
+    size_t ylocalsize = 1;
+    size_t ygridsize  = 1;
+
+    auto kernel = KernelInfo{};
+
+    kernel.kernel_file = "MIOpenPadConstantFwd.cpp";
+    kernel.kernel_name = "PadConstantFwdContiguous";
+
+    // TODO: Actually understand how to use this properly
```

Comment:
This is the part where the macro is passed when building the kernel. Therefore, assuming that it is a float tpye, "#define INPUT_TYPE float" is passed to the actual kernel.
In general, we support half, float, and bfloat16 data types, so the code is written like the example we provided.
In the case of bfloat16, it has the type of ushort inside HIP. Of course, the actual operation is performed after type conversion to float.
Please refer to https://github.com/ROCm/MIOpen/blob/develop/src/kernels/float_types.h for why we pass MIOPEN_USE_FP32 related macros here.